Sort icons before packing

scothis opened the pull request:
Currently, the output of sprite zero is dependent on the order of the
icons array passed in. This can lead to an unintentional mismatch of json
and png unless the input order is consistent between calls.

Now, the icons array is sorted by id before packing to ensure that the
same icon set will always produce the same sprite sheet.

tmcw commented:
Hm, should we sort by size as well as id? I'm not 100% sure but it may have an effect https://en.wikipedia.org/wiki/Bin_packing_problem#Analysis_of_approximate_algorithms

bhousel commented:
Here's [a better paper](http://clb.demon.fi/files/RectangleBinPack.pdf) comparing binpacking algorithms.  

shelf-pack isn't really ready yet for spritezero (see #22)
But if we ever do use it, that binpacker uses SHELF-BHF-WM, so presorting by icon height ascending, then by id, would pack most efficiently and result in a deterministic pack.

scothis commented:
Updated to sort by height and then id.

Also added a script to regenerate the test fixtures, ripped off from the spritezero-cli. I was having a horrible time trying to use the cli with a dev version of spritezero.
